kano_settings/system/advanced.py

```diff
@@ -415,13 +415,19 @@ def make_safesearch_config_file(config_file, opendns):
     else:
         # use google servers 
         servers = ', '.join(google_servers)
+
     import pycountry, json
-    rule = "cname ^(?!forcesafesearch)(.*).(?:google|youtube).(?:{}) to forcesafesearch.google.com using "+servers    
+
+    rule_google = "cname ^(?:www.google.)(?:{}) to forcesafesearch.google.com using " + google_servers
+    rule_youtube = "cname ^(?:www.youtube.|m.youtube.|youtubei.googleapis.|youtube.googleapis.|www.youtube-nocookie.)' \
```

Comment:
A little confused here - you start the string with a `"` and end it with a `'` only for it to continue on the next line with a `'` and end with a `"`. Methinks that this isn't quite right. Also if you are going to break the line, keep the following `(?:{})` with it so that the urls aren't broken up. Something like

```python
rule_youtube = "cname ^(?:www.youtube.|m.youtube.|youtubei.googleapis.|youtube.googleapis.|www.youtube-nocookie.)(?:{}) " \
    "to restrict.youtube.com using " + google_servers
```